### PR TITLE
Add DonationRepository::getDonationById

### PR DESCRIPTION
**Pull request by JeroenDeDauw**

### CODE REVIEW

File: src/DataAccess/DoctrineDonationRepository.php [hunk 1]
```diff
@@ -31,15 +33,17 @@ public function __construct( EntityManager $entityManager ) {
 	}
 
 	public function storeDonation( Donation $donation ) {
+		$doctrineDonation = $this->newDonationEntity( $donation );
+
 		try {
-			$this->entityManager->persist( $this->newDonationEntity( $donation ) );
+			$this->entityManager->persist($doctrineDonation  );
```
Comment (gbirke) [inline src/DataAccess/DoctrineDonationRepository.php]: missing space

Comment (JeroenDeDauw) [inline src/DataAccess/DoctrineDonationRepository.php]: lostinspacelogo

Seriously though, what happened with the CS rule that checked this...

File: src/DataAccess/DoctrineDonationRepository.php [hunk 2]
```diff
@@ -136,4 +140,117 @@ private function getDataFieldsFromAddress( PhysicalAddress $address ) {
...
+	}
+
+	private function getPersonalInfoFromEntity( DoctrineDonation $dd ): PersonalInfo {
+		$personalInfo = new PersonalInfo();
+
+		$personalInfo->setEmailAddress( $dd->getEmail() );
+		$personalInfo->setPersonName( $this->getPersonNameFromEntity( $dd ) );
+		$personalInfo->setPhysicalAddress( $this->getPhysicalAddressFromEntity( $dd ) );
+		$personalInfo->setPhysicalAddress( $this->getPhysicalAddressFromEntity( $dd ) );
+		$personalInfo->setPhysicalAddress( $this->getPhysicalAddressFromEntity( $dd ) );
```
Comment (KaiNissen) [inline src/DataAccess/DoctrineDonationRepository.php]: You are really insisting on setting the physical address.

Comment (JeroenDeDauw) [inline src/DataAccess/DoctrineDonationRepository.php]: Yeah. Should I add a comment to clarify this is just in case the first call does not work properly?

Comment (KaiNissen) [inline src/DataAccess/DoctrineDonationRepository.php]: You could also extract it into a method. Something like `hitThePersonalInfoWithLeadPipeUntilItFinallySetsTheAddress()`.

**Pull request merged**